src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/EducationAssessmentService.kt

```diff
@@ -0,0 +1,103 @@
+package uk.gov.justice.digital.hmpps.hmppsintegrationapi.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import jakarta.validation.ValidationException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.exception.EntityNotFoundException
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.exception.MessageFailedException
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.EducationAssessmentStatus
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.EducationAssessmentStatusChangeRequest
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.HmppsMessage
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.HmppsMessageEventType
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.HmppsMessageResponse
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.Response
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.UpstreamApiError
+import uk.gov.justice.hmpps.sqs.HmppsQueue
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+import uk.gov.justice.hmpps.sqs.eventTypeMessageAttributes
+import java.time.LocalDate
+
+@Service
+class EducationAssessmentService(
+  private val getPersonService: GetPersonService,
+  private val hmppsQueueService: HmppsQueueService,
+  private val objectMapper: ObjectMapper,
+) {
+  private val assessmentEventsQueue by lazy { hmppsQueueService.findByQueueId("assessmentevents") as HmppsQueue }
+  private val assessmentEventsQueueSqsClient by lazy { assessmentEventsQueue.sqsClient }
+  private val assessmentEventsQueueUrl by lazy { assessmentEventsQueue.queueUrl }
+
+  companion object {
+    private val logger: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun sendEducationAssessmentEvent(
+    hmppsId: String,
+    request: EducationAssessmentStatusChangeRequest,
+  ): Response<HmppsMessageResponse> {
+    val personResponse = getPersonService.getNomisNumber(hmppsId = hmppsId)
+
+    if (personResponse.hasError(UpstreamApiError.Type.ENTITY_NOT_FOUND)) {
```

Comment:
Typically we've been doing this conversion from error to exception at the controller layer. I'm not convinced that is a good pattern, so I won't request a change here.